test/functional/feature_config_args.py

```diff
@@ -186,12 +186,28 @@ def test_seed_peers(self):
         with self.nodes[0].assert_debug_log(expected_msgs=[
                 "Loaded 0 addresses from peers.dat",
                 "DNS seeding disabled",
-                "Adding fixed seeds as -dnsseed=0, -addnode is not provided and all -seednode(s) attempted\n",
+                "Adding fixed seeds as -dnsseed=0 (or clearnet connections are disabled via -onlynet), -addnode is not provided and all -seednode(s) attempted\n",
         ]):
             self.start_node(0, extra_args=['-dnsseed=0', '-fixedseeds=1'])
         assert time.time() - start < 60
         self.stop_node(0)
 
+        # No peers.dat exists, -dnsseed=1 but -onlynet=i2p
```

Comment:
If `dnsseed=1` is set, I would expect it to use DNS seeds no matter what the other options are...

Reply:
It's the default value, so many users will have it set without explicitely choosing it. Or do you suggest to have behavior conditional on whether it is set as a default value, or explicitly specified by the user?

But I can't see a point in `-dnsseed=1 -onlynet=i2p` - for me, it is essentially like `-noconnect`: no attempts for outgoing connections are made, because we'll load address from DNS seeds that we then refuse to connect to.

Reply:
>It's the default value, so many users will have it set without explicitely choosing it. Or do you suggest to have behavior conditional on whether it is set as a default value, or explicitly specified by the user?

Yes, like SoftSetArg already does for other similar situations

>But I can't see a point in -dnsseed=1 -onlynet=i2p - for me, it is essentially like -noconnect: no attempts for outgoing connections are made, because we'll load address from DNS seeds that we then refuse to connect to.

Perhaps not, but for tor it does make some sense.

Reply:
> Yes, like SoftSetArg already does for other similar situations

Ok, I understand it would have to be separate from the other SoftSetArg interactions, because the precondition onlynet is only evaluated during `AppInitMain`

> Perhaps not, but for tor it does make some sense.

Why? I tend to see the current behavior for Tor (adding the dns seed as an [addr fetch](https://github.com/bitcoin/bitcoin/blob/a6fc293c0a1f27ba1e573bfa16fd76d5f58988b2/src/net.cpp#L1457-L1459), and making a clearnet-connection to a random node it resolves to) as a privacy bug and not something a user who specified `-onlynet=onion` could possibly want to do on purpose.

Reply:
It's undesirable to just ignore the user's request if they explicitly specify `-dnsseed=1`. Perhaps it makes sense to give an InitError, or even just a warning?

Reply:
Since we'd ignore a user's request whatever we'd to in this situation (either `-dnseed=1` or `-onlynet=onion`) I like the suggestion of returning an InitError and not ignoring either of the two conflicting requests. Will adjust the PR to implement that.

Reply:
Done with the latest push.